### PR TITLE
Issue #27/ingest drug synonyms

### LINKED ISSUE #27: Ingest drug synonyms
-- Load drugbank.ca
-- Identify drug names and synonyms
-- Pair against clinical trial drug names
-- Write to neo4j
-- Run continuously via prefect.io

### PR DESCRIPTION
**Pull request by 007vasy**
The code is working e2e, the big questions is what is the best way to include the neo4j class created in this issue into the master one.

After that the URL parsing and enrichment could be done easily.

### CODE REVIEW

**Review by bechbd — DISMISSED** [review]

File: modules/Neo4jDataAccess.py [hunk 1]
```diff
@@ -2,18 +2,40 @@
...
+
 import enum
 
 from datetime import datetime
 import pandas as pd
 from neo4j import GraphDatabase, basic_auth
 from urllib.parse import urlparse
 import logging
-
+logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
```
Comment (lmeyerov) [inline modules/Neo4jDataAccess.py]: indiv files shouldn't be setting logging levels

File: modules/Neo4jDataAccess.py [hunk 2]
```diff
@@ -424,3 +446,146 @@ def __parse_urls(self, row, url_params, job_name, job_id=None):
...
+        node_merging_func = self._merge_node
+        with self._driver.session() as session:
+            logger.info("> Merging Urls Job is Started")
+            count_node = 0
+            prev_count_node = 0
+            
+            for url in urls:
+                node_type = "Url"
+                properties:dict = {"url":url}
+                url_id = session.write_transaction(node_merging_func, node_type, properties)
```
Comment (lmeyerov) [inline modules/Neo4jDataAccess.py]: is this using the `Url`  format w/ all the url props?

**Review by lmeyerov — CHANGES_REQUESTED** [review]
1. did you mean to rm `.gitignore` ?

2. is this creating a `:Url` node with all props? (see slack conv..)

File: modules/DrugSynonymDataToNeo4j.py [hunk 3]
```diff
@@ -14,7 +15,7 @@ def property_type_checker(property_value):
         elif isinstance(property_value,str):
             property_value = '''"''' + property_value.replace('"',r"\"") + '''"'''
         elif not property_value:
-            property_value = ""
+            property_value = "''"
```
Comment (lmeyerov) [inline modules/DrugSynonymDataToNeo4j.py:18]: @bechbd  not sure of best practices here wrt neo4j

File: modules/DrugSynonymDataToNeo4j.py [hunk 4]
```diff
@@ -31,55 +32,105 @@ def cypher_template_filler(cypher_template:str,data:dict) -> str:
...
                 properties:dict = study
-                session.write_transaction(node_merging_func, node_type, properties)
+                study_id = session.write_transaction(node_merging_func, node_type, properties)
+                self.study_triald_and_neo4j_id_pairs[study["trial_id"]] = study_id
                 count_node += 1
-                if count_node > prev_count_node + 100:
+                if count_node > prev_count_node + 1000:
                     prev_count_node = count_node
-                    logger.info("> {} nodes already imported".format(count_node)) 
+                    logger.info("> {} nodes already merged".format(count_node))
```
Comment (lmeyerov) [inline modules/DrugSynonymDataToNeo4j.py:103]: non-blocker, but generally better to do 
`logger.info("blah %s blah", myvar)` so when in production mode, string interpolation methods don't execute ( => faster), esp. good when in data loops like this

File: modules/DrugSynonymDataToNeo4j.py (hunk 3 above)
Comment (007vasy) [reply modules/DrugSynonymDataToNeo4j.py:18]: this is only if the property value is None

**Review by lmeyerov — APPROVED** [review]
IMO fine to land vs. drift

From here, we should work w/ @bechbd to get into actual neo4j then w/ @ZiyaoWei  as always-on prefect flow, and we can test/tweak further as needed through that

File: modules/DrugSynonymDataToNeo4j.py (hunk 4 above)
Comment (007vasy) [reply modules/DrugSynonymDataToNeo4j.py:103]: do you want me to update it right now?

Comment (lmeyerov) [reply modules/DrugSynonymDataToNeo4j.py:103]: Up to you. Fine merging as-is, I just like being helpful in code reviews as a mutual learning process :)

**Pull request merged**